### PR TITLE
Add DiscordBM/DiscordLogger to loggers list

### LINKED ISSUE #256: Add new Discord Logger to README
### Discord Logger

 * I've recently publicized the Discord-logger part of the private Discord stuff I'm using.

* For what it's worth, I've personally moved to the publicized version of the logger since a few days ago, and everything has been going smoothly. 

* We're soon going to use the logger in [Vapor community's Penny](https://github.com/vapor/penny-bot/pull/39). 

* The logger also has a [decent amount of tests](https://github.com/MahdiBM/DiscordBM/blob/main/Tests/DiscordBMTests/DiscordLogger.swift).

* It does take care of edge cases. For example, the logger:
  * Won't exceed Discord message-length limits. Will try to truncate if possible.
  * Won't exceed its Discord rate-limit when sending logs.
  * Will escape Discord-special characters before sending over to Discord.
 
  And much more ...
  
If you need to review the implementation, the logger-related parts of the library are present in the [DiscordLogger](https://github.com/MahdiBM/DiscordBM/tree/main/Sources/DiscordLogger) target and there is a [Discord-logger section available in the README](https://github.com/MahdiBM/DiscordBM/blob/discord-logger/README.md#discord-logger).

### PR DESCRIPTION
**Pull request by MahdiBM**
### Motivation:

Add [DiscordBM](https://github.com/MahdiBM/DiscordBM/blob/discord-logger/README.md#discord-logger) to the loggers list.

The description is:

> a Discord package containing a standalone [DiscordLogger library](https://github.com/MahdiBM/DiscordBM/blob/discord-logger/README.md#discord-logger)

Please do tell if you think "standalone" seems misleading 🤔 I'm just trying to mention that users don't need to depend on the full package just for Discord-logging.

Resolves #256

### Modifications:

Edit README logger-list table.

### Result:

More awareness about existence of a Swift package capable of logging to Discord.

### CODE REVIEW

**Comment by ktoso** [comment]
@swift-server-bot test this please

File: README.md [hunk 1]
```diff
@@ -85,6 +85,7 @@ As the API has just launched, not many implementations exist yet. If you are int
 | [ShivaHuang/swift-log-SwiftyBeaver](https://github.com/ShivaHuang/swift-log-SwiftyBeaver) | a logging backend for printing colored logging to Xcode console / file, or sending encrypted logging to [SwiftyBeaver](https://swiftybeaver.com) platform. |
 | [Apodini/swift-log-elk](https://github.com/Apodini/swift-log-elk) | a logging backend that formats, caches and sends log data to [elastic/logstash](https://github.com/elastic/logstash) |
 | [binaryscraping/swift-log-supabase](https://github.com/binaryscraping/swift-log-supabase) | a logging backend that sends log entries to [Supabase](https://github.com/supabase/supabase). |
+| [MahdiBM/DiscordBM](https://github.com/MahdiBM/DiscordBM) | a Discord package containing a standalone [DiscordLogger library](https://github.com/MahdiBM/DiscordBM/blob/discord-logger/README.md#discord-logger) |
```
Comment (ktoso) [inline README.md:88]: "a Discord package" sounds a bit weird, perhaps "a Discord client package"?

Comment (MahdiBM) [reply README.md:88]: Yeah it doesn't sound the best. I just wrote that based on what SPM calls them. The "DiscordLogger library" part also came from the fact that in Package.swift you do .library(name: "DiscordLogger"... couldn't also word that better.

Comment (MahdiBM) [reply README.md:88]: If you mean that it might sound like that Discord actually built the package, then i guess that's a valid point too.

**Comment by ktoso** [comment]
@swift-server-bot test this please

**Review by ktoso — APPROVED** [review]
Thanks!

**Comment by MahdiBM** [comment]
@ktoso I think we need another round of running tests here 👀

**Comment by ktoso** [comment]
Looks good now, thanks! Yep, we'll merge after the checks run

**Comment by ktoso** [comment]
@swift-server-bot test this please
